ICE: `attempt to add with overflow` emitter.rs

<!--
Thank you for finding an Internal Compiler Error! 🧊  If possible, try to provide
a minimal verifiable example. You can read "Rust Bug Minimization Patterns" for
how to create smaller examples.
http://blog.pnkfx.org/blog/2019/11/18/rust-bug-minimization-patterns/
-->

### Code
Note that this is debug assertion is space-sensi
```Rust
fn main() {
    {
        let mut mutex: libc() = std::mem::zeroed(
            file.as_raw_fd(),
            0,
            0,
            libc::SYNC_FILE_RANGE_WAIT_BEFORE,
        );
    }
}

```


### Meta
<!--
If you're using the stable version of the compiler, you should also check if the
bug also exists in the beta or nightly versions.
-->

`rustc --version --verbose`:
```
<version>
```

### Error output

```
error[E0433]: failed to resolve: use of undeclared crate or module `libc`
 --> treereduce.out:7:13
  |
7 |             libc::SYNC_FILE_RANGE_WAIT_BEFORE,
  |             ^^^^ use of undeclared crate or module `libc`

error[E0412]: cannot find type `libc` in this scope
 --> treereduce.out:3:24
  |
3 |         let mut mutex: libc() = std::mem::zeroed(
  |                        ^^^^ not found in this scope

error[E0423]: expected value, found macro `file`
 --> treereduce.out:4:13
  |
4 |             file.as_raw_fd(),
  |             ^^^^ not a value

error[E0061]: this function takes 0 arguments but 4 arguments were supplied
   --> treereduce.out:3:33
    |
3   |         let mut mutex: libc() = std::mem::zeroed(
    |                                 ^^^^^^^^^^^^^^^^
4   |             file.as_raw_fd(),
    |             ---------------- unexpected argument
5   |             0,
    |             - unexpected argument of type `{integer}`
6   |             0,
    |             - unexpected argument of type `{integer}`
7   |             libc::SYNC_FILE_RANGE_WAIT_BEFORE,
    |             --------------------------------- unexpected argument
    |
note: function defined here
   --> /home/matthias/vcs/github/rust_debug_assertions/library/core/src/mem/mod.rs:643:15
    |
643 | pub unsafe fn zeroed<T>() -> T {
    |               ^^^^^^
```

<!--
Include a backtrace in the code block by setting `RUST_BACKTRACE=1` in your
environment. E.g. `RUST_BACKTRACE=1 cargo build`.
-->
<details><summary><strong>Backtrace</strong></summary>
<p>

```
thread 'rustc' panicked at 'attempt to add with overflow', compiler/rustc_errors/src/emitter.rs:2266:17
stack backtrace:
   0:     0x7f558dd0d254 - std::backtrace_rs::backtrace::libunwind::trace::h154b4ffb8ddaa4c5
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1:     0x7f558dd0d254 - std::backtrace_rs::backtrace::trace_unsynchronized::h80c3570785f36849
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x7f558dd0d254 - std::sys_common::backtrace::_print_fmt::h4c724079b81989bf
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/sys_common/backtrace.rs:65:5
   3:     0x7f558dd0d254 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::hb8e6266f21fedb2e
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/sys_common/backtrace.rs:44:22
   4:     0x7f558ddeb468 - core::fmt::write::h2dac34aae88eac3a
                               at /home/matthias/vcs/github/rust_debug_assertions/library/core/src/fmt/mod.rs:1254:17
   5:     0x7f558dd369af - std::io::Write::write_fmt::he4a6f86c0870d055
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/io/mod.rs:1684:15
   6:     0x7f558dd0d055 - std::sys_common::backtrace::_print::hd9fa4da5fb35e3c2
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/sys_common/backtrace.rs:47:5
   7:     0x7f558dd0d055 - std::sys_common::backtrace::print::h96adbffa1fffcf04
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/sys_common/backtrace.rs:34:9
   8:     0x7f558dd38cc4 - std::panicking::default_hook::{{closure}}::hb425055b1e88d07e
   9:     0x7f558dd38922 - std::panicking::default_hook::h096318d450721a10
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/panicking.rs:290:9
  10:     0x7f55908217a5 - <alloc[df9d1ca7e15cbd48]::boxed::Box<dyn for<'a, 'b> core[56d2be079082b2f7]::ops::function::Fn<(&'a core[56d2be079082b2f7]::panic::panic_info::PanicInfo<'b>,), Output = ()> + core[56d2be079082b2f7]::marker::Send + core[56d2be079082b2f7]::marker::Sync> as core[56d2be079082b2f7]::ops::function::Fn<(&core[56d2be079082b2f7]::panic::panic_info::PanicInfo,)>>::call
                               at /home/matthias/vcs/github/rust_debug_assertions/library/alloc/src/boxed.rs:2002:9
  11:     0x7f55908217a5 - rustc_driver_impl[9a61a7d58ba23a35]::DEFAULT_HOOK::{closure#0}::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_driver_impl/src/lib.rs:1185:17
  12:     0x7f558dd39489 - <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call::hd5e684618c375fd7
                               at /home/matthias/vcs/github/rust_debug_assertions/library/alloc/src/boxed.rs:2002:9
  13:     0x7f558dd39489 - std::panicking::rust_panic_with_hook::he6a6c9655719b7f7
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/panicking.rs:696:13
  14:     0x7f558dd0dec2 - std::panicking::begin_panic_handler::{{closure}}::h9c8e0e7be4b0ecd5
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/panicking.rs:581:13
  15:     0x7f558dd0d336 - std::sys_common::backtrace::__rust_end_short_backtrace::hae9ffc8460a9e61b
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/sys_common/backtrace.rs:150:18
  16:     0x7f558dd38fb2 - rust_begin_unwind
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/panicking.rs:579:5
  17:     0x7f558ddc8873 - core::panicking::panic_fmt::h3d0a8f81bb9ae67d
                               at /home/matthias/vcs/github/rust_debug_assertions/library/core/src/panicking.rs:64:14
  18:     0x7f558ddc890d - core::panicking::panic::h8315a6df2d01c70b
                               at /home/matthias/vcs/github/rust_debug_assertions/library/core/src/panicking.rs:114:5
  19:     0x7f559094a93e - <rustc_errors[52e33a5894953d58]::emitter::EmitterWriter>::draw_code_line
  20:     0x7f559094819c - <rustc_errors[52e33a5894953d58]::emitter::EmitterWriter>::emit_suggestion_default
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_errors/src/emitter.rs:1951:17
  21:     0x7f5590943877 - <rustc_errors[52e33a5894953d58]::emitter::EmitterWriter>::emit_messages_default
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_errors/src/emitter.rs:2147:49
  22:     0x7f5590943877 - <rustc_errors[52e33a5894953d58]::emitter::EmitterWriter as rustc_errors[52e33a5894953d58]::emitter::Emitter>::emit_diagnostic
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_errors/src/emitter.rs:554:9
  23:     0x7f559092aa79 - <rustc_errors[52e33a5894953d58]::HandlerInner>::emit_diagnostic::{closure#2}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_errors/src/lib.rs:1401:17
  24:     0x7f55913b51a9 - rustc_interface[ce7826790f4f2a50]::callbacks::track_diagnostic::{closure#0}::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_interface/src/callbacks.rs:41:53
  25:     0x7f55913b51a9 - rustc_middle[41bf02a7f05d410d]::ty::context::tls::tlv::with_tlv::<rustc_interface[ce7826790f4f2a50]::callbacks::track_diagnostic::{closure#0}::{closure#0}, ()>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:94:9
  26:     0x7f55913b51a9 - rustc_middle[41bf02a7f05d410d]::ty::context::tls::enter_context::<rustc_interface[ce7826790f4f2a50]::callbacks::track_diagnostic::{closure#0}::{closure#0}, ()>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:114:5
  27:     0x7f55913b51a9 - rustc_interface[ce7826790f4f2a50]::callbacks::track_diagnostic::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_interface/src/callbacks.rs:41:20
  28:     0x7f55913b51a9 - rustc_middle[41bf02a7f05d410d]::ty::context::tls::with_context_opt::<rustc_interface[ce7826790f4f2a50]::callbacks::track_diagnostic::{closure#0}, ()>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:131:18
  29:     0x7f55913b51a9 - rustc_interface[ce7826790f4f2a50]::callbacks::track_diagnostic
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_interface/src/callbacks.rs:31:5
  30:     0x7f5590929c4e - <rustc_errors[52e33a5894953d58]::HandlerInner>::emit_diagnostic
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_errors/src/lib.rs:1370:9
  31:     0x7f559098cf96 - <rustc_errors[52e33a5894953d58]::Handler>::emit_diagnostic
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_errors/src/lib.rs:1121:9
  32:     0x7f559098cf96 - <rustc_span[4d46095f68ead720]::ErrorGuaranteed as rustc_errors[52e33a5894953d58]::diagnostic_builder::EmissionGuarantee>::diagnostic_builder_emit_producing_guarantee
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_errors/src/diagnostic_builder.rs:169:28
  33:     0x7f5590e31143 - <rustc_errors[52e33a5894953d58]::diagnostic_builder::DiagnosticBuilder<rustc_span[4d46095f68ead720]::ErrorGuaranteed>>::emit
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_errors/src/diagnostic_builder.rs:505:9
  34:     0x7f5590e31143 - <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::report_arg_errors
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs:1253:9
  35:     0x7f5590e2bcf6 - <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_argument_types
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs:453:13
  36:     0x7f5590df064a - <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::confirm_builtin_call
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/callee.rs:460:9
  37:     0x7f5590deecb8 - <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_call
  38:     0x7f5590e848e3 - <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_expr_kind
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:329:45
  39:     0x7f5590e09df4 - <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:236:18
  40:     0x7f5590e09df4 - stacker[a1b2ff0363358e05]::maybe_grow::<rustc_middle[41bf02a7f05d410d]::ty::Ty, <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}>
                               at /home/matthias/.cargo/registry/src/github.com-1ecc6299db9ec823/stacker-0.1.15/src/lib.rs:55:9
  41:     0x7f5590e09df4 - rustc_data_structures[409a719ca8def432]::stack::ensure_sufficient_stack::<rustc_middle[41bf02a7f05d410d]::ty::Ty, <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_data_structures/src/stack.rs:17:5
  42:     0x7f5590e09df4 - <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:232:18
  43:     0x7f5590e83bb2 - <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_expr_with_expectation
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:186:9
  44:     0x7f5590e336d0 - <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_expr_with_hint
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:138:9
  45:     0x7f5590e336d0 - <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_expr_coercable_to_type
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:128:18
  46:     0x7f5590e336d0 - <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_decl_initializer
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs:1387:13
  47:     0x7f5590e336d0 - <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_decl
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs:1398:27
  48:     0x7f5590e33b54 - <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_decl_local
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs:1431:9
  49:     0x7f5590e33b54 - <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_stmt
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs:1448:17
  50:     0x7f5590e34326 - <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_block_with_expected::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs:1521:17
  51:     0x7f5590e34326 - <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::with_breakable_ctxt::<<rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_block_with_expected::{closure#0}, ()>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/fn_ctxt/_impl.rs:1489:22
  52:     0x7f5590e34326 - <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_block_with_expected
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs:1519:26
  53:     0x7f5590e844ed - <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_expr_kind
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:328:41
  54:     0x7f5590e09df4 - <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:236:18
  55:     0x7f5590e09df4 - stacker[a1b2ff0363358e05]::maybe_grow::<rustc_middle[41bf02a7f05d410d]::ty::Ty, <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}>
                               at /home/matthias/.cargo/registry/src/github.com-1ecc6299db9ec823/stacker-0.1.15/src/lib.rs:55:9
  56:     0x7f5590e09df4 - rustc_data_structures[409a719ca8def432]::stack::ensure_sufficient_stack::<rustc_middle[41bf02a7f05d410d]::ty::Ty, <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_data_structures/src/stack.rs:17:5
  57:     0x7f5590e09df4 - <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:232:18
  58:     0x7f5590e83bb2 - <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_expr_with_expectation
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:186:9
  59:     0x7f5590e3435b - <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_block_with_expected::{closure#0}::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs:1526:50
  60:     0x7f5590e3435b - <core[56d2be079082b2f7]::option::Option<&&rustc_hir[978887b891d924c0]::hir::Expr>>::map::<rustc_middle[41bf02a7f05d410d]::ty::Ty, <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_block_with_expected::{closure#0}::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/library/core/src/option.rs:1137:29
  61:     0x7f5590e3435b - <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_block_with_expected::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs:1526:32
  62:     0x7f5590e3435b - <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::with_breakable_ctxt::<<rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_block_with_expected::{closure#0}, ()>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/fn_ctxt/_impl.rs:1489:22
  63:     0x7f5590e3435b - <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_block_with_expected
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs:1519:26
  64:     0x7f5590e844ed - <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_expr_kind
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:328:41
  65:     0x7f5590e09df4 - <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:236:18
  66:     0x7f5590e09df4 - stacker[a1b2ff0363358e05]::maybe_grow::<rustc_middle[41bf02a7f05d410d]::ty::Ty, <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}>
                               at /home/matthias/.cargo/registry/src/github.com-1ecc6299db9ec823/stacker-0.1.15/src/lib.rs:55:9
  67:     0x7f5590e09df4 - rustc_data_structures[409a719ca8def432]::stack::ensure_sufficient_stack::<rustc_middle[41bf02a7f05d410d]::ty::Ty, <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_data_structures/src/stack.rs:17:5
  68:     0x7f5590e09df4 - <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:232:18
  69:     0x7f5590e83bb2 - <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_expr_with_expectation
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:186:9
  70:     0x7f5590e0bbb6 - <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_expr_with_hint
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:138:9
  71:     0x7f5590e0bbb6 - <rustc_hir_typeck[5012b823f920620b]::fn_ctxt::FnCtxt>::check_return_expr
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:825:30
  72:     0x7f5591014437 - rustc_hir_typeck[5012b823f920620b]::check::check_fn
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/check.rs:122:9
  73:     0x7f559100df14 - rustc_hir_typeck[5012b823f920620b]::typeck_with_fallback::<rustc_hir_typeck[5012b823f920620b]::typeck::{closure#0}>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/lib.rs:231:9
  74:     0x7f559100df14 - rustc_hir_typeck[5012b823f920620b]::typeck_with_fallback::<rustc_hir_typeck[5012b823f920620b]::typeck::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/lib.rs:191:36
  75:     0x7f559100df14 - rustc_hir_typeck[5012b823f920620b]::typeck
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/lib.rs:172:9
  76:     0x7f5592447feb - <rustc_query_impl[cc0295f81e1a13f0]::queries::typeck as rustc_query_system[c7e7491ac393df84]::query::config::QueryConfig<rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>>::compute
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:519:21
  77:     0x7f5592447feb - rustc_query_system[c7e7491ac393df84]::query::plumbing::execute_job::<rustc_query_impl[cc0295f81e1a13f0]::queries::typeck, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:440:71
  78:     0x7f5592447feb - rustc_middle[41bf02a7f05d410d]::ty::context::tls::tlv::with_tlv::<rustc_query_system[c7e7491ac393df84]::query::plumbing::execute_job<rustc_query_impl[cc0295f81e1a13f0]::queries::typeck, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}, &rustc_middle[41bf02a7f05d410d]::ty::typeck_results::TypeckResults>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:94:9
  79:     0x7f5592447feb - rustc_middle[41bf02a7f05d410d]::ty::context::tls::enter_context::<rustc_query_system[c7e7491ac393df84]::query::plumbing::execute_job<rustc_query_impl[cc0295f81e1a13f0]::queries::typeck, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}, &rustc_middle[41bf02a7f05d410d]::ty::typeck_results::TypeckResults>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:114:5
  80:     0x7f5592447feb - <rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt as rustc_query_system[c7e7491ac393df84]::query::QueryContext>::start_query::<&rustc_middle[41bf02a7f05d410d]::ty::typeck_results::TypeckResults, rustc_query_system[c7e7491ac393df84]::query::plumbing::execute_job<rustc_query_impl[cc0295f81e1a13f0]::queries::typeck, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:127:13
  81:     0x7f5592447feb - rustc_middle[41bf02a7f05d410d]::ty::context::tls::with_related_context::<<rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt as rustc_query_system[c7e7491ac393df84]::query::QueryContext>::start_query<&rustc_middle[41bf02a7f05d410d]::ty::typeck_results::TypeckResults, rustc_query_system[c7e7491ac393df84]::query::plumbing::execute_job<rustc_query_impl[cc0295f81e1a13f0]::queries::typeck, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, &rustc_middle[41bf02a7f05d410d]::ty::typeck_results::TypeckResults>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:164:9
  82:     0x7f5592447feb - rustc_middle[41bf02a7f05d410d]::ty::context::tls::with_context::<rustc_middle[41bf02a7f05d410d]::ty::context::tls::with_related_context<<rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt as rustc_query_system[c7e7491ac393df84]::query::QueryContext>::start_query<&rustc_middle[41bf02a7f05d410d]::ty::typeck_results::TypeckResults, rustc_query_system[c7e7491ac393df84]::query::plumbing::execute_job<rustc_query_impl[cc0295f81e1a13f0]::queries::typeck, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, &rustc_middle[41bf02a7f05d410d]::ty::typeck_results::TypeckResults>::{closure#0}, &rustc_middle[41bf02a7f05d410d]::ty::typeck_results::TypeckResults>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:142:36
  83:     0x7f5592447feb - rustc_middle[41bf02a7f05d410d]::ty::context::tls::with_context_opt::<rustc_middle[41bf02a7f05d410d]::ty::context::tls::with_context<rustc_middle[41bf02a7f05d410d]::ty::context::tls::with_related_context<<rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt as rustc_query_system[c7e7491ac393df84]::query::QueryContext>::start_query<&rustc_middle[41bf02a7f05d410d]::ty::typeck_results::TypeckResults, rustc_query_system[c7e7491ac393df84]::query::plumbing::execute_job<rustc_query_impl[cc0295f81e1a13f0]::queries::typeck, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, &rustc_middle[41bf02a7f05d410d]::ty::typeck_results::TypeckResults>::{closure#0}, &rustc_middle[41bf02a7f05d410d]::ty::typeck_results::TypeckResults>::{closure#0}, &rustc_middle[41bf02a7f05d410d]::ty::typeck_results::TypeckResults>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:131:18
  84:     0x7f5592447feb - rustc_middle[41bf02a7f05d410d]::ty::context::tls::with_context::<rustc_middle[41bf02a7f05d410d]::ty::context::tls::with_related_context<<rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt as rustc_query_system[c7e7491ac393df84]::query::QueryContext>::start_query<&rustc_middle[41bf02a7f05d410d]::ty::typeck_results::TypeckResults, rustc_query_system[c7e7491ac393df84]::query::plumbing::execute_job<rustc_query_impl[cc0295f81e1a13f0]::queries::typeck, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, &rustc_middle[41bf02a7f05d410d]::ty::typeck_results::TypeckResults>::{closure#0}, &rustc_middle[41bf02a7f05d410d]::ty::typeck_results::TypeckResults>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:142:5
  85:     0x7f5592447feb - rustc_middle[41bf02a7f05d410d]::ty::context::tls::with_related_context::<<rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt as rustc_query_system[c7e7491ac393df84]::query::QueryContext>::start_query<&rustc_middle[41bf02a7f05d410d]::ty::typeck_results::TypeckResults, rustc_query_system[c7e7491ac393df84]::query::plumbing::execute_job<rustc_query_impl[cc0295f81e1a13f0]::queries::typeck, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, &rustc_middle[41bf02a7f05d410d]::ty::typeck_results::TypeckResults>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:155:5
  86:     0x7f5592447feb - <rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt as rustc_query_system[c7e7491ac393df84]::query::QueryContext>::start_query::<&rustc_middle[41bf02a7f05d410d]::ty::typeck_results::TypeckResults, rustc_query_system[c7e7491ac393df84]::query::plumbing::execute_job<rustc_query_impl[cc0295f81e1a13f0]::queries::typeck, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:112:9
  87:     0x7f5592447feb - rustc_query_system[c7e7491ac393df84]::query::plumbing::execute_job::<rustc_query_impl[cc0295f81e1a13f0]::queries::typeck, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:440:17
  88:     0x7f5592447feb - rustc_query_system[c7e7491ac393df84]::query::plumbing::try_execute_query::<rustc_query_impl[cc0295f81e1a13f0]::queries::typeck, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:382:44
  89:     0x7f55926ce8d5 - rustc_query_system[c7e7491ac393df84]::query::plumbing::get_query::<rustc_query_impl[cc0295f81e1a13f0]::queries::typeck, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:814:36
  90:     0x7f55926ce8d5 - stacker[a1b2ff0363358e05]::maybe_grow::<(&rustc_middle[41bf02a7f05d410d]::ty::typeck_results::TypeckResults, core[56d2be079082b2f7]::option::Option<rustc_query_system[c7e7491ac393df84]::dep_graph::graph::DepNodeIndex>), rustc_query_system[c7e7491ac393df84]::query::plumbing::get_query<rustc_query_impl[cc0295f81e1a13f0]::queries::typeck, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}>
                               at /home/matthias/.cargo/registry/src/github.com-1ecc6299db9ec823/stacker-0.1.15/src/lib.rs:55:9
  91:     0x7f55926ce8d5 - rustc_data_structures[409a719ca8def432]::stack::ensure_sufficient_stack::<(&rustc_middle[41bf02a7f05d410d]::ty::typeck_results::TypeckResults, core[56d2be079082b2f7]::option::Option<rustc_query_system[c7e7491ac393df84]::dep_graph::graph::DepNodeIndex>), rustc_query_system[c7e7491ac393df84]::query::plumbing::get_query<rustc_query_impl[cc0295f81e1a13f0]::queries::typeck, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_data_structures/src/stack.rs:17:5
  92:     0x7f55926ce8d5 - rustc_query_system[c7e7491ac393df84]::query::plumbing::get_query::<rustc_query_impl[cc0295f81e1a13f0]::queries::typeck, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:814:9
  93:     0x7f55926ce8d5 - <rustc_query_impl[cc0295f81e1a13f0]::Queries as rustc_middle[41bf02a7f05d410d]::ty::query::QueryEngine>::typeck::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:805:17
  94:     0x7f55926ce8d5 - <rustc_query_impl[cc0295f81e1a13f0]::Queries as rustc_middle[41bf02a7f05d410d]::ty::query::QueryEngine>::typeck
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/lib.rs:47:1
  95:     0x7f5590f408bb - std[a36a78ba7a755d70]::panicking::try::do_call::<core[56d2be079082b2f7]::panic::unwind_safe::AssertUnwindSafe<rustc_data_structures[409a719ca8def432]::sync::par_for_each_in<&[rustc_span[4d46095f68ead720]::def_id::LocalDefId], <rustc_middle[41bf02a7f05d410d]::hir::map::Map>::par_body_owners<rustc_hir_typeck[5012b823f920620b]::typeck_item_bodies::{closure#0}>::{closure#0}>::{closure#0}::{closure#0}>, ()>
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/panicking.rs:487:40
  96:     0x7f5590f408bb - std[a36a78ba7a755d70]::panicking::try::<(), core[56d2be079082b2f7]::panic::unwind_safe::AssertUnwindSafe<rustc_data_structures[409a719ca8def432]::sync::par_for_each_in<&[rustc_span[4d46095f68ead720]::def_id::LocalDefId], <rustc_middle[41bf02a7f05d410d]::hir::map::Map>::par_body_owners<rustc_hir_typeck[5012b823f920620b]::typeck_item_bodies::{closure#0}>::{closure#0}>::{closure#0}::{closure#0}>>
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/panicking.rs:451:19
  97:     0x7f5590f408bb - std[a36a78ba7a755d70]::panic::catch_unwind::<core[56d2be079082b2f7]::panic::unwind_safe::AssertUnwindSafe<rustc_data_structures[409a719ca8def432]::sync::par_for_each_in<&[rustc_span[4d46095f68ead720]::def_id::LocalDefId], <rustc_middle[41bf02a7f05d410d]::hir::map::Map>::par_body_owners<rustc_hir_typeck[5012b823f920620b]::typeck_item_bodies::{closure#0}>::{closure#0}>::{closure#0}::{closure#0}>, ()>
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/panic.rs:140:14
  98:     0x7f5590f408bb - rustc_data_structures[409a719ca8def432]::sync::par_for_each_in::<&[rustc_span[4d46095f68ead720]::def_id::LocalDefId], <rustc_middle[41bf02a7f05d410d]::hir::map::Map>::par_body_owners<rustc_hir_typeck[5012b823f920620b]::typeck_item_bodies::{closure#0}>::{closure#0}>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_data_structures/src/sync.rs:156:33
  99:     0x7f5590f408bb - <core[56d2be079082b2f7]::slice::iter::Iter<rustc_span[4d46095f68ead720]::def_id::LocalDefId> as core[56d2be079082b2f7]::iter::traits::iterator::Iterator>::for_each::<rustc_data_structures[409a719ca8def432]::sync::par_for_each_in<&[rustc_span[4d46095f68ead720]::def_id::LocalDefId], <rustc_middle[41bf02a7f05d410d]::hir::map::Map>::par_body_owners<rustc_hir_typeck[5012b823f920620b]::typeck_item_bodies::{closure#0}>::{closure#0}>::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/library/core/src/slice/iter/macros.rs:201:21
 100:     0x7f5590f408bb - rustc_data_structures[409a719ca8def432]::sync::par_for_each_in::<&[rustc_span[4d46095f68ead720]::def_id::LocalDefId], <rustc_middle[41bf02a7f05d410d]::hir::map::Map>::par_body_owners<rustc_hir_typeck[5012b823f920620b]::typeck_item_bodies::{closure#0}>::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_data_structures/src/sync.rs:155:13
 101:     0x7f559100ad33 - <rustc_middle[41bf02a7f05d410d]::hir::map::Map>::par_body_owners::<rustc_hir_typeck[5012b823f920620b]::typeck_item_bodies::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/hir/map/mod.rs:491:9
 102:     0x7f559100ad33 - rustc_hir_typeck[5012b823f920620b]::typeck_item_bodies
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/lib.rs:156:5
 103:     0x7f55923528a0 - <rustc_query_impl[cc0295f81e1a13f0]::queries::typeck_item_bodies as rustc_query_system[c7e7491ac393df84]::query::config::QueryConfig<rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>>::compute
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:519:21
 104:     0x7f55923528a0 - rustc_query_system[c7e7491ac393df84]::query::plumbing::execute_job::<rustc_query_impl[cc0295f81e1a13f0]::queries::typeck_item_bodies, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:440:71
 105:     0x7f55923528a0 - rustc_middle[41bf02a7f05d410d]::ty::context::tls::tlv::with_tlv::<rustc_query_system[c7e7491ac393df84]::query::plumbing::execute_job<rustc_query_impl[cc0295f81e1a13f0]::queries::typeck_item_bodies, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}, ()>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:94:9
 106:     0x7f55923528a0 - rustc_middle[41bf02a7f05d410d]::ty::context::tls::enter_context::<rustc_query_system[c7e7491ac393df84]::query::plumbing::execute_job<rustc_query_impl[cc0295f81e1a13f0]::queries::typeck_item_bodies, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}, ()>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:114:5
 107:     0x7f55923528a0 - <rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt as rustc_query_system[c7e7491ac393df84]::query::QueryContext>::start_query::<(), rustc_query_system[c7e7491ac393df84]::query::plumbing::execute_job<rustc_query_impl[cc0295f81e1a13f0]::queries::typeck_item_bodies, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:127:13
 108:     0x7f55923528a0 - rustc_middle[41bf02a7f05d410d]::ty::context::tls::with_related_context::<<rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt as rustc_query_system[c7e7491ac393df84]::query::QueryContext>::start_query<(), rustc_query_system[c7e7491ac393df84]::query::plumbing::execute_job<rustc_query_impl[cc0295f81e1a13f0]::queries::typeck_item_bodies, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, ()>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:164:9
 109:     0x7f55923528a0 - rustc_middle[41bf02a7f05d410d]::ty::context::tls::with_context::<rustc_middle[41bf02a7f05d410d]::ty::context::tls::with_related_context<<rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt as rustc_query_system[c7e7491ac393df84]::query::QueryContext>::start_query<(), rustc_query_system[c7e7491ac393df84]::query::plumbing::execute_job<rustc_query_impl[cc0295f81e1a13f0]::queries::typeck_item_bodies, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, ()>::{closure#0}, ()>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:142:36
 110:     0x7f55923528a0 - rustc_middle[41bf02a7f05d410d]::ty::context::tls::with_context_opt::<rustc_middle[41bf02a7f05d410d]::ty::context::tls::with_context<rustc_middle[41bf02a7f05d410d]::ty::context::tls::with_related_context<<rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt as rustc_query_system[c7e7491ac393df84]::query::QueryContext>::start_query<(), rustc_query_system[c7e7491ac393df84]::query::plumbing::execute_job<rustc_query_impl[cc0295f81e1a13f0]::queries::typeck_item_bodies, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, ()>::{closure#0}, ()>::{closure#0}, ()>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:131:18
 111:     0x7f55923528a0 - rustc_middle[41bf02a7f05d410d]::ty::context::tls::with_context::<rustc_middle[41bf02a7f05d410d]::ty::context::tls::with_related_context<<rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt as rustc_query_system[c7e7491ac393df84]::query::QueryContext>::start_query<(), rustc_query_system[c7e7491ac393df84]::query::plumbing::execute_job<rustc_query_impl[cc0295f81e1a13f0]::queries::typeck_item_bodies, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, ()>::{closure#0}, ()>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:142:5
 112:     0x7f55923528a0 - rustc_middle[41bf02a7f05d410d]::ty::context::tls::with_related_context::<<rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt as rustc_query_system[c7e7491ac393df84]::query::QueryContext>::start_query<(), rustc_query_system[c7e7491ac393df84]::query::plumbing::execute_job<rustc_query_impl[cc0295f81e1a13f0]::queries::typeck_item_bodies, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, ()>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:155:5
 113:     0x7f55923528a0 - <rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt as rustc_query_system[c7e7491ac393df84]::query::QueryContext>::start_query::<(), rustc_query_system[c7e7491ac393df84]::query::plumbing::execute_job<rustc_query_impl[cc0295f81e1a13f0]::queries::typeck_item_bodies, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:112:9
 114:     0x7f55923528a0 - rustc_query_system[c7e7491ac393df84]::query::plumbing::execute_job::<rustc_query_impl[cc0295f81e1a13f0]::queries::typeck_item_bodies, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:440:17
 115:     0x7f55923528a0 - rustc_query_system[c7e7491ac393df84]::query::plumbing::try_execute_query::<rustc_query_impl[cc0295f81e1a13f0]::queries::typeck_item_bodies, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:382:44
 116:     0x7f55926ce02e - rustc_query_system[c7e7491ac393df84]::query::plumbing::get_query::<rustc_query_impl[cc0295f81e1a13f0]::queries::typeck_item_bodies, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:814:36
 117:     0x7f55926ce02e - stacker[a1b2ff0363358e05]::maybe_grow::<((), core[56d2be079082b2f7]::option::Option<rustc_query_system[c7e7491ac393df84]::dep_graph::graph::DepNodeIndex>), rustc_query_system[c7e7491ac393df84]::query::plumbing::get_query<rustc_query_impl[cc0295f81e1a13f0]::queries::typeck_item_bodies, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}>
                               at /home/matthias/.cargo/registry/src/github.com-1ecc6299db9ec823/stacker-0.1.15/src/lib.rs:55:9
 118:     0x7f55926ce02e - rustc_data_structures[409a719ca8def432]::stack::ensure_sufficient_stack::<((), core[56d2be079082b2f7]::option::Option<rustc_query_system[c7e7491ac393df84]::dep_graph::graph::DepNodeIndex>), rustc_query_system[c7e7491ac393df84]::query::plumbing::get_query<rustc_query_impl[cc0295f81e1a13f0]::queries::typeck_item_bodies, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_data_structures/src/stack.rs:17:5
 119:     0x7f55926ce02e - rustc_query_system[c7e7491ac393df84]::query::plumbing::get_query::<rustc_query_impl[cc0295f81e1a13f0]::queries::typeck_item_bodies, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:814:9
 120:     0x7f55926ce02e - <rustc_query_impl[cc0295f81e1a13f0]::Queries as rustc_middle[41bf02a7f05d410d]::ty::query::QueryEngine>::typeck_item_bodies::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:805:17
 121:     0x7f55926ce02e - <rustc_query_impl[cc0295f81e1a13f0]::Queries as rustc_middle[41bf02a7f05d410d]::ty::query::QueryEngine>::typeck_item_bodies
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/lib.rs:47:1
 122:     0x7f5590b35109 - <rustc_middle[41bf02a7f05d410d]::ty::query::TyCtxtAt>::typeck_item_bodies
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/query.rs:380:29
 123:     0x7f5590b35109 - <rustc_middle[41bf02a7f05d410d]::ty::context::TyCtxt>::typeck_item_bodies
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/query.rs:366:35
 124:     0x7f5590b35109 - rustc_hir_analysis[5d5c63228a24b74c]::check_crate::{closure#7}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_analysis/src/lib.rs:501:50
 125:     0x7f5590b35109 - <rustc_data_structures[409a719ca8def432]::profiling::VerboseTimingGuard>::run::<(), rustc_hir_analysis[5d5c63228a24b74c]::check_crate::{closure#7}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_data_structures/src/profiling.rs:723:9
 126:     0x7f5590b35109 - <rustc_session[222c0bb8eb24a151]::session::Session>::time::<(), rustc_hir_analysis[5d5c63228a24b74c]::check_crate::{closure#7}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_session/src/utils.rs:10:50
 127:     0x7f5590ca53f9 - rustc_hir_analysis[5d5c63228a24b74c]::check_crate
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_analysis/src/lib.rs:501:5
 128:     0x7f559137aa9d - rustc_interface[ce7826790f4f2a50]::passes::analysis
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_interface/src/passes.rs:758:5
 129:     0x7f559244d9c7 - <rustc_query_impl[cc0295f81e1a13f0]::queries::analysis as rustc_query_system[c7e7491ac393df84]::query::config::QueryConfig<rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>>::compute
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:519:21
 130:     0x7f559244d9c7 - rustc_query_system[c7e7491ac393df84]::query::plumbing::execute_job::<rustc_query_impl[cc0295f81e1a13f0]::queries::analysis, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:440:71
 131:     0x7f559244d9c7 - rustc_middle[41bf02a7f05d410d]::ty::context::tls::tlv::with_tlv::<rustc_query_system[c7e7491ac393df84]::query::plumbing::execute_job<rustc_query_impl[cc0295f81e1a13f0]::queries::analysis, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}, core[56d2be079082b2f7]::result::Result<(), rustc_span[4d46095f68ead720]::ErrorGuaranteed>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:94:9
 132:     0x7f559244d9c7 - rustc_middle[41bf02a7f05d410d]::ty::context::tls::enter_context::<rustc_query_system[c7e7491ac393df84]::query::plumbing::execute_job<rustc_query_impl[cc0295f81e1a13f0]::queries::analysis, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}, core[56d2be079082b2f7]::result::Result<(), rustc_span[4d46095f68ead720]::ErrorGuaranteed>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:114:5
 133:     0x7f559244d9c7 - <rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt as rustc_query_system[c7e7491ac393df84]::query::QueryContext>::start_query::<core[56d2be079082b2f7]::result::Result<(), rustc_span[4d46095f68ead720]::ErrorGuaranteed>, rustc_query_system[c7e7491ac393df84]::query::plumbing::execute_job<rustc_query_impl[cc0295f81e1a13f0]::queries::analysis, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:127:13
 134:     0x7f559244d9c7 - rustc_middle[41bf02a7f05d410d]::ty::context::tls::with_related_context::<<rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt as rustc_query_system[c7e7491ac393df84]::query::QueryContext>::start_query<core[56d2be079082b2f7]::result::Result<(), rustc_span[4d46095f68ead720]::ErrorGuaranteed>, rustc_query_system[c7e7491ac393df84]::query::plumbing::execute_job<rustc_query_impl[cc0295f81e1a13f0]::queries::analysis, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, core[56d2be079082b2f7]::result::Result<(), rustc_span[4d46095f68ead720]::ErrorGuaranteed>>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:164:9
 135:     0x7f559244d9c7 - rustc_middle[41bf02a7f05d410d]::ty::context::tls::with_context::<rustc_middle[41bf02a7f05d410d]::ty::context::tls::with_related_context<<rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt as rustc_query_system[c7e7491ac393df84]::query::QueryContext>::start_query<core[56d2be079082b2f7]::result::Result<(), rustc_span[4d46095f68ead720]::ErrorGuaranteed>, rustc_query_system[c7e7491ac393df84]::query::plumbing::execute_job<rustc_query_impl[cc0295f81e1a13f0]::queries::analysis, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, core[56d2be079082b2f7]::result::Result<(), rustc_span[4d46095f68ead720]::ErrorGuaranteed>>::{closure#0}, core[56d2be079082b2f7]::result::Result<(), rustc_span[4d46095f68ead720]::ErrorGuaranteed>>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:142:36
 136:     0x7f559244d9c7 - rustc_middle[41bf02a7f05d410d]::ty::context::tls::with_context_opt::<rustc_middle[41bf02a7f05d410d]::ty::context::tls::with_context<rustc_middle[41bf02a7f05d410d]::ty::context::tls::with_related_context<<rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt as rustc_query_system[c7e7491ac393df84]::query::QueryContext>::start_query<core[56d2be079082b2f7]::result::Result<(), rustc_span[4d46095f68ead720]::ErrorGuaranteed>, rustc_query_system[c7e7491ac393df84]::query::plumbing::execute_job<rustc_query_impl[cc0295f81e1a13f0]::queries::analysis, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, core[56d2be079082b2f7]::result::Result<(), rustc_span[4d46095f68ead720]::ErrorGuaranteed>>::{closure#0}, core[56d2be079082b2f7]::result::Result<(), rustc_span[4d46095f68ead720]::ErrorGuaranteed>>::{closure#0}, core[56d2be079082b2f7]::result::Result<(), rustc_span[4d46095f68ead720]::ErrorGuaranteed>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:131:18
 137:     0x7f559244d9c7 - rustc_middle[41bf02a7f05d410d]::ty::context::tls::with_context::<rustc_middle[41bf02a7f05d410d]::ty::context::tls::with_related_context<<rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt as rustc_query_system[c7e7491ac393df84]::query::QueryContext>::start_query<core[56d2be079082b2f7]::result::Result<(), rustc_span[4d46095f68ead720]::ErrorGuaranteed>, rustc_query_system[c7e7491ac393df84]::query::plumbing::execute_job<rustc_query_impl[cc0295f81e1a13f0]::queries::analysis, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, core[56d2be079082b2f7]::result::Result<(), rustc_span[4d46095f68ead720]::ErrorGuaranteed>>::{closure#0}, core[56d2be079082b2f7]::result::Result<(), rustc_span[4d46095f68ead720]::ErrorGuaranteed>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:142:5
 138:     0x7f559244d9c7 - rustc_middle[41bf02a7f05d410d]::ty::context::tls::with_related_context::<<rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt as rustc_query_system[c7e7491ac393df84]::query::QueryContext>::start_query<core[56d2be079082b2f7]::result::Result<(), rustc_span[4d46095f68ead720]::ErrorGuaranteed>, rustc_query_system[c7e7491ac393df84]::query::plumbing::execute_job<rustc_query_impl[cc0295f81e1a13f0]::queries::analysis, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, core[56d2be079082b2f7]::result::Result<(), rustc_span[4d46095f68ead720]::ErrorGuaranteed>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:155:5
 139:     0x7f559244d9c7 - <rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt as rustc_query_system[c7e7491ac393df84]::query::QueryContext>::start_query::<core[56d2be079082b2f7]::result::Result<(), rustc_span[4d46095f68ead720]::ErrorGuaranteed>, rustc_query_system[c7e7491ac393df84]::query::plumbing::execute_job<rustc_query_impl[cc0295f81e1a13f0]::queries::analysis, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:112:9
 140:     0x7f559244d9c7 - rustc_query_system[c7e7491ac393df84]::query::plumbing::execute_job::<rustc_query_impl[cc0295f81e1a13f0]::queries::analysis, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:440:17
 141:     0x7f559244d9c7 - rustc_query_system[c7e7491ac393df84]::query::plumbing::try_execute_query::<rustc_query_impl[cc0295f81e1a13f0]::queries::analysis, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:382:44
 142:     0x7f559269ad7e - rustc_query_system[c7e7491ac393df84]::query::plumbing::get_query::<rustc_query_impl[cc0295f81e1a13f0]::queries::analysis, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:814:36
 143:     0x7f559269ad7e - stacker[a1b2ff0363358e05]::maybe_grow::<(core[56d2be079082b2f7]::result::Result<(), rustc_span[4d46095f68ead720]::ErrorGuaranteed>, core[56d2be079082b2f7]::option::Option<rustc_query_system[c7e7491ac393df84]::dep_graph::graph::DepNodeIndex>), rustc_query_system[c7e7491ac393df84]::query::plumbing::get_query<rustc_query_impl[cc0295f81e1a13f0]::queries::analysis, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}>
                               at /home/matthias/.cargo/registry/src/github.com-1ecc6299db9ec823/stacker-0.1.15/src/lib.rs:55:9
 144:     0x7f559269ad7e - rustc_data_structures[409a719ca8def432]::stack::ensure_sufficient_stack::<(core[56d2be079082b2f7]::result::Result<(), rustc_span[4d46095f68ead720]::ErrorGuaranteed>, core[56d2be079082b2f7]::option::Option<rustc_query_system[c7e7491ac393df84]::dep_graph::graph::DepNodeIndex>), rustc_query_system[c7e7491ac393df84]::query::plumbing::get_query<rustc_query_impl[cc0295f81e1a13f0]::queries::analysis, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_data_structures/src/stack.rs:17:5
 145:     0x7f559269ad7e - rustc_query_system[c7e7491ac393df84]::query::plumbing::get_query::<rustc_query_impl[cc0295f81e1a13f0]::queries::analysis, rustc_query_impl[cc0295f81e1a13f0]::plumbing::QueryCtxt>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:814:9
 146:     0x7f559269ad7e - <rustc_query_impl[cc0295f81e1a13f0]::Queries as rustc_middle[41bf02a7f05d410d]::ty::query::QueryEngine>::analysis::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:805:17
 147:     0x7f559269ad7e - <rustc_query_impl[cc0295f81e1a13f0]::Queries as rustc_middle[41bf02a7f05d410d]::ty::query::QueryEngine>::analysis
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/lib.rs:47:1
 148:     0x7f55908235f4 - <rustc_middle[41bf02a7f05d410d]::ty::query::TyCtxtAt>::analysis
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/query.rs:380:29
 149:     0x7f55908235f4 - <rustc_middle[41bf02a7f05d410d]::ty::context::TyCtxt>::analysis
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/query.rs:366:35
 150:     0x7f55908235f4 - rustc_driver_impl[9a61a7d58ba23a35]::run_compiler::{closure#1}::{closure#2}::{closure#4}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_driver_impl/src/lib.rs:386:52
 151:     0x7f55908235f4 - <rustc_middle[41bf02a7f05d410d]::ty::context::GlobalCtxt>::enter::<rustc_driver_impl[9a61a7d58ba23a35]::run_compiler::{closure#1}::{closure#2}::{closure#4}, core[56d2be079082b2f7]::result::Result<(), rustc_span[4d46095f68ead720]::ErrorGuaranteed>>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context.rs:560:37
 <snip>

error: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.70.0-dev running on x86_64-unknown-linux-gnu

query stack during panic:
#0 [typeck] type-checking `main`
#1 [typeck_item_bodies] type-checking all item bodies
#2 [analysis] running analysis passes on this crate
end of query stack
error: aborting due to 3 previous errors
```

</p>
</details>

--- a/compiler/rustc_errors/src/lib.rs
+++ b/compiler/rustc_errors/src/lib.rs
@@ -331,7 +331,7 @@ impl CodeSuggestion {
                     });
                     buf.push_str(&part.snippet);
                     let cur_hi = sm.lookup_char_pos(part.span.hi());
-                    if cur_hi.line == cur_lo.line {
+                    if cur_hi.line == cur_lo.line && !part.snippet.is_empty() {
                         // Account for the difference between the width of the current code and the
                         // snippet being suggested, so that the *later* suggestions are correctly
                         // aligned on the screen.

--- a/tests/ui/suggestions/issue-109396.rs
+++ b/tests/ui/suggestions/issue-109396.rs
@@ -1,0 +1,12 @@
+fn main() {
+    {
+        let mut mutex = std::mem::zeroed(
+            //~^ ERROR this function takes 0 arguments but 4 arguments were supplied
+            file.as_raw_fd(),
+            //~^ ERROR expected value, found macro `file`
+            0,
+            0,
+            0,
+        );
+    }
+}

--- a/tests/ui/suggestions/issue-109396.stderr
+++ b/tests/ui/suggestions/issue-109396.stderr
@@ -1,0 +1,34 @@
+error[E0423]: expected value, found macro `file`
+  --> $DIR/issue-109396.rs:5:13
+   |
+LL |             file.as_raw_fd(),
+   |             ^^^^ not a value
+
+error[E0061]: this function takes 0 arguments but 4 arguments were supplied
+  --> $DIR/issue-109396.rs:3:25
+   |
+LL |         let mut mutex = std::mem::zeroed(
+   |                         ^^^^^^^^^^^^^^^^
+LL |
+LL |             file.as_raw_fd(),
+   |             ---------------- unexpected argument
+LL |
+LL |             0,
+   |             - unexpected argument of type `{integer}`
+LL |             0,
+   |             - unexpected argument of type `{integer}`
+LL |             0,
+   |             - unexpected argument of type `{integer}`
+   |
+note: function defined here
+  --> $SRC_DIR/core/src/mem/mod.rs:LL:COL
+help: remove the extra arguments
+   |
+LL -             file.as_raw_fd(),
+LL +             ,
+   |
+
+error: aborting due to 2 previous errors
+
+Some errors have detailed explanations: E0061, E0423.
+For more information about an error, try `rustc --explain E0061`.

